lib/msf/core/post/windows/priv.rb

```diff
@@ -124,23 +124,18 @@ def is_system?
   # if running on a system that does not have UAC
   #
   def is_uac_enabled?
-    uac = false
-    winversion = session.sys.config.sysinfo['OS']
+    return false if is_system?
 
-    if winversion =~ /Windows (Vista|7|8|2008|2012|10)/
-      unless is_system?
-        begin
-          enable_lua = registry_getvaldata(
-              'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System',
-              'EnableLUA'
-          )
-          uac = (enable_lua == 1)
-        rescue Rex::Post::Meterpreter::RequestError => e
-          print_error("Error Checking if UAC is Enabled: #{e.class} #{e}")
-        end
-      end
-    end
-    return uac
+    winversion = session.sys.config.sysinfo['OS']
+    return false unless winversion =~ /Windows (Vista|7|8|10|2008|2012|2016)/
+
+    registry_getvaldata(
+      'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System',
+      'EnableLUA'
+    ).eql?(1)
```

Comment:
I get what you are trying to do here, but `#eql?` is best reserved for use in data structures like `Hash` and `Set` where you really care about type and value. Are you worried that `1 == ...` might get confused as an assignment? Ending an interrogator (ie. `?`) method with a boolean expression is common practice in Ruby, and since we have to do the `registry_getvaldata` call across multiple lines having the constant on the left ensures its visibility. If you want to unambiguously get across the idea that "this is a comparison, and yes, we are using this value" you are free to add an explicit `return`, compare against `0x01`, or even both.

Reply:
No